### ISSUE TITLE
Enhancement: Transform Zero validator should allow Deltas

### ISSUE THREAD

**Issue by alainxi** [ynput/OpenPype#5582]
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Please describe the feature you have in mind and explain what the current shortcomings are?

In Blender, the "Transform zero" validator wants us to apply a freeze transform, this is useful.
But there are 2 ways to achieve that goal, and this validator wants us to use the method that cannot be used in most productions.

The method we need to use (but is refused by the validator) :
(from inside Blender) Object-->Apply-->**All Transforms to Deltas**
It transfers the transforms to the **Deltas** of objects (it places the pivot at the center of the **object**, so you **can** use it for animation).

The method that Ayon's Blender validator wants us to use :
(from inside Blender) Object-->Apply-->**All Transforms**
It transfers the transforms to the **datas** of objects (it places the pivot at the center of the **world**, so you **cannot** use it for animation).

### How would you imagine the implementation of the feature?

Here is the line :

https://github.com/ynput/OpenPype/blob/529ab6ca9bf0de6656fa11ba5f4e38af4b048049/openpype/hosts/blender/plugins/publish/validate_transform_zero.py#L35

It is currently using "matrix_basis", I hope there is a way to do the same thing while allowing deltas.
https://docs.blender.org/api/current/bpy.types.Object.html#bpy.types.Object.matrix_basis

If not possible using matrix, a brute force solution would be to test translation and rotation to be 0,0,0 and scale to be 1,1,1

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the enhancement request.

**Comment by alainxi** [comment]
Sorry, I don't find a wauy to add the "Host: Blender" label on this issue.

**Comment by alainxi** [comment]
In case some studios prefer the current behaviour, it would be even better to have a button that allows to select the behavior we prefer.

**Comment by BigRoy** [comment]
> In case some studios prefer the current behaviour, it would be even better to have a button that allows to select the behavior we prefer (allow Deltas or not).

This would be a studio settings > project settings, yes?

**Comment by alainxi** [comment]
> This would be a studio settings > project settings, yes?

Yes, I mean to have a new button here, to choose between "Allow Deltas" or not :
-  **Studio**  Settings-->Blender-->Publish Plugins-->Validators-->Validate Transform zero
-  **Project** Settings-->Blender-->Publish Plugins-->Validators-->Validate Transform zero

Currently it only has those 2 options :      "Active", "Optional".

**Comment by LiborBatek** [comment]
@simonebarbieri could be an easy fix, no?

**Comment by BigRoy** [comment]
I recall actually implementing this some time ago or testing it - but I can't find the code :D

I had [this comment](https://github.com/BigRoy/OpenPype/blob/4ddc3de82c82480a3630e9ff02f178a125ed8e71/openpype/hosts/blender/plugins/publish/validate_transform_zero.py#L66-L72). Apparently my code is already quite different [from `develop`](https://github.com/ynput/OpenPype/blob/develop/openpype/hosts/blender/plugins/publish/validate_transform_zero.py) so it might be a decent starting point.

Looking back now, my focus seemed mostly on adding a `repair` method at the time.

**Comment by alainxi** [comment]
I would like to add that this feature is not as important for us as it was.
Because we changed our realtime tools to allow working with Ayon's current behaviour.
